nikola/plugins/compile/ipynb.py

```diff
@@ -85,35 +84,123 @@ def read_metadata(self, post, file_metadata_regexp=None, unslugify_titles=False,
     def create_post(self, path, **kw):
         content = kw.pop('content', None)
         onefile = kw.pop('onefile', False)
+        kernel = kw.pop('ipython_kernel', None)
         # is_page is not needed to create the file
         kw.pop('is_page', False)
 
+        metadata = {}
+        metadata.update(self.default_metadata)
+        metadata.update(kw)
+
         makedirs(os.path.dirname(path))
+
+        if IPython.version_info[0] >= 3:
+            nb = nbformat.v4.new_notebook()
+            nb["cells"] = [nbformat.v4.new_code_cell(content)]
+        else:
+            nb = nbformat.v3.nbbase.new_notebook()
+            nb["cells"] = [nbformat.v3.nbbase.new_code_cell(content)]
+
         if onefile:
-            raise Exception('The one-file format is not supported by this compiler.')
+            nb["metadata"]["nikola"] = metadata
+
+        if kernel is not None:
+            nb["metadata"]["kernelspec"] = ipython_kernel_spec[kernel]
+            nb["metadata"]["language_info"] = ipython_language_info[kernel]
+
         with io.open(path, "w+", encoding="utf8") as fd:
-            if not content.startswith("Write your"):
```

Comment:
> This was an intentional feature: users could create an ipynb outside of Nikola and just import it by using the -f option. And we want to keep it in place.

In fact, rethinking this, I would like the users to use `nikola new_post --import path/to/ipynb`, but I have to investigate a little bit how `--import` works

> You will have to refactor this function. Due to the onefile changes, you might have to load the notebook, add meta (and nothing else) and save it back.

Will look into it...